quickwit/quickwit-codegen/src/codegen.rs

```diff
@@ -412,7 +484,16 @@ fn generate_client(context: &CodegenContext) -> TokenStream {
                         Output = Result<http::Response<hyper::Body>, quickwit_common::tower::BoxError>,
                     > + Send + 'static,
             {
-                #client_name::new(#grpc_client_adapter_name::new(#grpc_client_package_name::#grpc_client_name::new(channel)))
+                let (_, num_connections_watcher) = tokio::sync::watch::channel(1);
+                let adapter = #grpc_client_adapter_name::new(#grpc_client_package_name::#grpc_client_name::new(channel), num_connections_watcher);
+                Self::new(adapter)
+            }
+
+            pub fn from_balanced_channel<K: std::hash::Hash + Eq + Send + Clone + 'static>(balanced_channel: quickwit_common::tower::BalanceChannel<K>) -> #client_name
```

Comment:
I would keep only the from_balanced_channel.

Reply:
Yes, let's just keep the `from_balanced_channel`. The other one is harmful.

Reply:
I finally kept the `from_channel` too.